### ISSUE TITLE
Assigning IPv6 addresses with VirtualBox provider on Windows host fails

### ISSUE THREAD

**Issue by mrcage** [hashicorp/vagrant#10142]
I tried to use an IPv6 address with Vagrant, but starting the box fails.

### Vagrant version
Vagrant 2.1.2
VirtualBox 5.2.18

### Host operating system
Windows 7 64-bit, Windows 10 64-bit

### Guest operating system
Ubuntu Bionic 64-bit

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/bionic64"
  config.vm.network "private_network", ip: "fd12:3456:789a:1::2"
end
```

### Debug output
```
==> default: Clearing any previously set network interfaces...
There was an error while executing `VBoxManage`, a CLI used by Vagrant
for controlling VirtualBox. The command and stderr is shown below.

Command: ["hostonlyif", "ipconfig", "VirtualBox Host-Only Ethernet Adapter #3", "--ipv6", "fd12:3456:789a:1::1", "--netmasklengthv6", "64"]

Stderr: VBoxManage.exe: error: IPv6 setting is not supported for this adapter
```

### Expected behavior
The IPv6 address should be assigned to the Host-Only adapter in VirtualBox, and as well inside the host operating system, and the startup process should continue.

### Actual behavior
The startup process fails

### Steps to reproduce
1. vagrant up.

VirtualBox seems to not be able to configure IPv6 addresses properly on Windows, the input boxes are greyed out in the configuration dialog (see https://imgur.com/htJOTqr). I can configure the address manually using Windows network adapter settings.

**Comment by sarangjo** [comment]
Seeing the same issues with the same environment. Is there any known workaround?

**Comment by Bilge** [comment]
This is a big problem.

**Comment by soapy1** [comment]
Hi there,
Looks like this issue is blocked on VirtualBox not supporting IPv6. 💻 🌎

**Comment by filiprem-fb** [comment]
@soapy1 VirtualBox supports ipv6, as you can see here: https://www.virtualbox.org/manual/ch08.html 
The issue is on Vagrant side not being able to enable IPv6 on vbox instances. 
Suggestion: reopen this please :-)

**Comment by mrcage** [comment]
Yes agree. The issue is not fixed, as Vbox clearly supports IPv6, and I am even using it on Windows, with manual configuration of the network adapter in windows network settings; but the settings in virtualbox are grayed-out, and the CLI API returns an error, so somehow there is something missing here...

**Comment by 7331eraCkcuFetHwoD** [comment]
2022 wow incredible job maintaining your product...…. (emoji through up and eating it again only to through it up again)

**Comment by b01** [comment]
There is a work around. You can use Vagrant to inject a `netplan` config to apply the IPv6 address. Please see how I did it here https://github.com/b01/learning/blob/main/kubernetes/samples/Vagrantfile#L85

That is how I got around this issue.

The problem is that Vagrant and VirtualBox have been killed by the companies that own them, albeit maintained enough to work on the latest versions of OSes with a few enhancements. However, they are so good that they still work better than most of the newer solutions out there.
